### PR TITLE
LibC: Implement time formatting functions, partially support timezone

### PR DESCRIPTION
**Pull request by howar6hill**
Also fix a bug in mktime() caused by forgetting to check if the present year
is a leap year.

### CODE REVIEW

File: Libraries/LibC/time.cpp [hunk 1]
```diff
@@ -106,53 +107,222 @@ time_t mktime(struct tm* tm)
...
+            if (destination_pos + 1 > static_cast<int>(max_size) - 1)
+                return 0;
+            destination[destination_pos++] = format[i];
+        } else if (++i < format_len) {
+            char component[30];
+
+            switch (format[i]) {
+            case 'a':
+                snprintf(component, sizeof component, "%s", wday_short_names[tm->tm_wday]);
+                break;
```
Comment (awesomekling) [inline Libraries/LibC/time.cpp]: Let's avoid C-style programming and use `StringBuilder` for this task instead. Something like this:

```cpp
StringBuilder builder;
...
case 'a':
    builder.append(wday_short_names[tm->tm_wday]);
    break;
...
case 'T':
    builder.appendf("%02d:%02d:%02d", ...);
    break;
...
```

**Comment by howar6hill** [comment]
Switched to StringBuilder.

File: Libraries/LibC/time.cpp [hunk 2]
```diff
@@ -106,53 +109,217 @@ time_t mktime(struct tm* tm)
...
+            case '%':
+                builder.append('%');
+                break;
+            default:
+                return 0;
+            }
+        }
+        if (builder.length() > max_size - 1)
+                return 0;
```
Comment (awesomekling) [inline Libraries/LibC/time.cpp]: Weird indentation here.

File: Libraries/LibC/time.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ struct tm {
     int tm_isdst; /* Daylight saving time */
 };
 
-extern long timezone;
+extern long timezone; // The difference in seconds between UTC and local time
```
Comment (awesomekling) [inline Libraries/LibC/time.h]: Let's not use C++ style comments ( `//` ) in C library headers.

Comment (howar6hill) [reply Libraries/LibC/time.h]: Fixed.

**Pull request merged**